pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -329,12 +403,45 @@ def sample(self, observable, shot_range=None, bin_size=None, counts=False):
     def expval(self, observable, shot_range=None, bin_size=None):
         if observable.name in [
             "Projector",
-            "Hamiltonian",
-            "SparseHamiltonian",
+            "Hermitian",
         ]:
             self.syncD2H()
             return super().expval(observable, shot_range=shot_range, bin_size=bin_size)
 
+        if observable.name in ["SparseHamiltonian"]:
+            CSR_SparseHamiltonian = observable.data[0].tocsr(copy=True)
+            return self._gpu_state.ExpectationValue(
+                CSR_SparseHamiltonian.indptr,
+                CSR_SparseHamiltonian.indices,
+                CSR_SparseHamiltonian.data,
+            )
+
+        if observable.name in ["Hamiltonian"]:
```

Comment:
`Hamiltonian` can have `Hermitian` objects in its `ops`. Do we support this case? In addition, it may be less verbose to serialize the object and use it in C++ (as in Lightning `master` does currently).

Reply:
Good idea. @multiphaseCFD may be worth looking back on this and matching the Lightning implementation.

Reply:
Thanks @chaeyeunpark and @mlxd ! Let me check it.

Reply:
Hermitian ops now raise an error when used in H.